lib/view_component/collection.rb

```diff
@@ -0,0 +1,54 @@
+
+# frozen_string_literal: true
+
+module ViewComponent
+  class Collection
+    def initialize(component, object = nil, options)
+      @component = component
+      @object = object
+      @options = options
+    end
+
+    def render_in(view_context, &block)
+      as = as_variable(@component.virtual_path, @options)
+      collection = collection_variable(@object, @options)
+      args = @options.except(:collection, :as)
+
+      collection.map do |item|
+        args[as] = item
```

Comment:
Probably a bit more defensive to create a copy of the args here just on the off chance that the Component manipulates them. Could just do `args.merge(as: item)`

Reply:
Agreed and implemented (slightly different implementation since the key is actually dynamic).